roles/nameserver/README.rst

```diff
@@ -0,0 +1,170 @@
+nameserver
+==========
+
+This role is used to set up and configure a very basic **internal** BIND DNS master server.
+
+This role has only been tested on CentOS 7.2 using BIND9.
+
+It does the following:
+
+- Installs and updates necessary packages
+- Enables and configures firewalld
+- Manages named.conf and BIND daemon config
+- Manages forward and reverse DNS records
+
+Prerequisites
++++++++++++++
+
+- CentOS 7.2
+
+Variables
++++++++++
+Most variables are defined in ``roles/nameserver/defaults/main.yml`` and values are chosen to support our Sepia_ lab.  They can be overridden in the ``secrets`` repo.
+
+A list of packages to install that is specific to the role.  The list is defined in ``roles/nameserver/vars/packages.yml``::
+
+    packages: []
+
+BIND main configuration directory.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_dir: "/var/named"
+
+BIND main configuration file.  This is the default CentOS path.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_file: "/etc/named.conf"
+
+BIND data directory.  named debug output and statistics are stored here.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_data_dir: "/var/named/data"
+
+Port BIND should listen on.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_listen_port: 53
+
+Interface(s) BIND should listen on.  This defaults to listen on all IPv4 interfaces Ansible detects for the nameserver.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_listen_iface:
+      - 127.0.0.1
+      - "{{ ansible_all_ipv4_addresses[0] }}"
+
+Path to BIND zone files.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_zones_path: "/var/named/zones"
+
+named daemon options.  Writes to ``/etc/sysconfig/named``.  Defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_daemon_opts: []
+
+    # Example for IPv4 support only:
+    named_conf_daemon_opts: "-4"
+
+The following variables are self-explanatory and written to each zone file's SOA resource record.  They're defined in ``roles/nameserver/defaults/main.yml``::
+
+    named_conf_ttl: 3600
```

Comment:
It might be slightly clearer to name these "named_conf_soa_ttl", just to make it clear in the code and in the variable file that they are for the SOA record only.  The doc couldn't be clearer, but for those times you're editing and aren't looking at the doc...